flux-tests/tests/pos/surface/ptr.rs

```diff
@@ -3,15 +3,23 @@
 
 use std::ptr;
 
-// #[flux::trusted]
 pub fn test0(vec: Vec<i32>) -> *const i32 {
     vec.as_ptr()
 }
 
-// #[flux::trusted] // pointer
-// #[flux::sig(fn (self: &mut VecDeque<T,A>[@me], off: usize{ off < me.cap }, value: T) )]
 pub fn buffer_write(vec: &mut Vec<i32>, off: usize, value: i32) {
     unsafe {
         ptr::write(vec.as_mut_ptr().add(off), value);
     }
 }
+
+pub fn get_mut(vec: &mut Vec<i32>, off: usize) -> Option<&mut i32> {
+    unsafe { Some(&mut *vec.as_mut_ptr().add(off)) }
+}
+
+pub unsafe fn copy_nonoverlapping(vec: &mut Vec<i32>, dst: usize, src: usize, len: usize) {
+    unsafe {
+        ptr::copy_nonoverlapping(vec.as_mut_ptr().add(src), vec.as_mut_ptr().add(dst), len);
+    }
+}
+
```

Comment:
Let's create a file with simple tests something like:

```rust
// MutToConstPointer
unsafe fn test00(p: *mut i32) {
    let x = p as *const i32
}

// Read from a raw pointer
unsafe fn test01(p: *[mut|const] i32) -> i32 {
     *p
}

// Pointer subtyping
unsafe fn test01(p: *[mut|const] i32) -> *[mut|const] i32 {
     p
}

// Borrow from a pointer
unsafe fn test01(p: *[mut|const] i32) -> &[mut] i32 {
     &mut *p
}
```

Reply:
Ok! I added a new test file with the various legal (?) combinations